render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
```

Comment:
### 1
it's not about size of the buffer/pointer in ctx -- it's more about when you're ready to start sending first bytes to `ResponseWriter` and thus to the client

I want to support this for example:
```go
render.Respond(r, w, slowS3FileIOReader) // io.Reader
render.Respond(r, w, slowChannelOfData) // chan Data
```
- this should start streaming bytes to the client right away
- and most importantly, it should **block the handler**, so you can use `defer` in the handler for clean-up
- if you don't, you'll end up buffering lot of data in the context, or you'll loose defer functionality in handlers

### 2
`generic pager` code can live in our custom `Respond()` function logic - there's no need to wait for a middleware layer down the chain on a way back from the handler to do the same thing.

### 3
This is how stdlib works too. Imho, no need to be more fancy. I don't see any problem with overriding the headers:
```go
w.Header().Set("Header", "first value")
w.Header().Set("Header", "no, sorry, this is the right value")
```